File: data/readable_format/fortran/file932.f90
! This version is unchanged, provided for context. Unit tests will be a separate program.
program file_example
  implicit none

  integer :: ierror

  open (unit=13, file="test.dat", status="new", action="write", iostat=ierror)
  if (ierror /= 0) then
    print*, "Failed to open test.dat"
    stop
  end if

  write (unit=13, fmt=*) "Hello world!"
  close (unit=13)
end program file_example

File: data/readable_format/cpp/file932.cpp
#include <iostream>
#include <fstream>

int main() {
    std::ofstream file("test.dat", std::ios::out); // Open the file for writing.

    if (!file.is_open()) { // Check if the file was successfully opened.
        std::cerr << "Failed to open test.dat" << std::endl;
        return 1; // Return with error code.
    }

    file << "Hello world!" << std::endl; // Write to the file.

    file.close(); // Close the file.

    return 0; // Return success.
}